src/new-components/modal/modal.tsx

```diff
@@ -248,13 +248,15 @@ export function ModalHeader({
             >
                 <Columns space="large" alignY="center">
                     <Column width="auto">{children}</Column>
-                    <Column width="content" exceptionallySetClassName={styles.buttonContainer}>
-                        {typeof button !== 'boolean' ? (
-                            button
-                        ) : button === true ? (
-                            <ModalCloseButton aria-label="Close modal" autoFocus={false} />
-                        ) : null}
-                    </Column>
+                    {button === false || button === null || button === undefined ? null : (
+                        <Column width="content" exceptionallySetClassName={styles.buttonContainer}>
```

Comment:
Could you add a test about this in the `modal.test.tsx` file (under the `describe('ModalHeader',` section maybe).

I'm not sure how to check this in the test, though. Specifically the part about how to check if this `Column` container is rendered, since it's just a div, it can be hard to address.

I suspect that in jest tests the CSS modules class names are used as is. So this container should have `class="buttonContainer"` in the resulting HTML.

```js
expect(document.querySelector('.buttonContainer')).not.toBeInTheDocument()
```

You can also add a `data-testid` attribute if that makes it easier.

Reply:
I previously skimmed the test suite for this component and was under the impression there was a test case that tested the hidden button, but I was wrong.

I've added the test case as per your suggestion.